module/zfs/spa_config.c

```diff
@@ -167,14 +167,9 @@ spa_config_write(spa_config_dirent_t *dp, nvlist_t *nvl)
 	/*
 	 * Pack the configuration into a buffer.
 	 */
-	VERIFY(nvlist_size(nvl, &buflen, NV_ENCODE_XDR) == 0);
-
-	buf = vmem_alloc(buflen, KM_SLEEP);
+	buf = fnvlist_pack(nvl, &buflen);
 	temp = kmem_zalloc(MAXPATHLEN, KM_SLEEP);
```

Comment:
That's a good idea...  and now that you mention it it appears we've already done it in commit 77aef6f60ea29f6d3769addc778db6328ac85755.  So actually, this code is good and we don't need to change anything in this patch.